happi/cli.py

```diff
@@ -1146,6 +1154,60 @@ def edit_config():
         subprocess.run([os.environ.get("EDITOR", "vi"), config_filepath])
 
 
+@config.command()
+@click.option("--overwrite/--no-overwrite", "overwrite", default=False,
+              help="Overwrite existing config.")
+def init(overwrite):
+    """Create configuration file with default options."""
+
+    # find config_filepath
+    try:
+        config_filepath = Path(happi.client.Client.find_config())
+        if not overwrite:
+            click.echo("Found existing config file at:")
+            click.echo(f"  {config_filepath}")
+            click.echo("Stopping! Use --overwrite to destroy this config file.")
+            return
+    except OSError:
+        config_filepath = Path(platformdirs.user_config_dir("happi")) / "happi.cfg"
+    click.echo("Creating new config file at:")
+    click.echo(f"  {config_filepath}")
+
+    # find database_filepath
+    database_filepath = Path(platformdirs.user_data_dir("happi")) / "db.json"
```

Comment:
This assumes the JSON backend, which is generally fine. I think for future-proofing, it might be wise to add a `backend` argument here that defaults to `"json"` and raises `NotImplementedError` if it's not `"json"`.

Reply:
Seems like a wise idea to have the option `--json`. By implementing it as a `click.Choice` users will never be able to enter anything except the options given (currently only `json`).

My "hot take" is that it's probably never going to be easy to just guess what config should look like for database backends... too many site-specific choices are inevitable. But perhaps it could go down a different codepath where it asked some questions.